newrelic-agent/src/main/java/com/newrelic/agent/tracers/AbstractTracer.java

```diff
@@ -231,6 +232,24 @@ public boolean trackChildThreads() {
         return this.trackChildThreads;
     }
 
+    @Override
+    public void setTrackCallbackRunnable(boolean shouldTrack) {
+        this.trackCallBackRunnable = shouldTrack;
+    }
+
+    @Override
+    public boolean trackCallbackRunnable() {
+        if (!this.trackCallBackRunnable) {
```

Comment:
If you factor out the guts of the `if` into a new private method called 'parentTrackCallbackRunnable()` or similar, then this collapses to a one-liner like
```
    @Override
    public boolean trackCallbackRunnable() {
        return this.trackCallBackRunnable || this.parentTrackCallbackRunnable();
    }
```